src/main/java/fruitshop/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,24 @@
+package fruitshop.strategy;
+
+import fruitshop.model.FruitTransaction;
+import fruitshop.model.Storage;
+
+public class BalanceOperation implements OperationHandler {
+    public static final String READ_FILE_PATH = "src/main/resources/reportToRead.csv";
+    private final Storage storage;
+
+    public BalanceOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handleTransaction(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() >= 0) {
+            storage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
```

Comment:
According to the task checklist, you should check the result balance before saving it in the Storage - it should be positive. You're currently checking if it's non-negative (greater than or equal to 0), which is correct, but please ensure that this condition is indeed intended in the context of your application.